packages/cloud/connection_details/components/copy_input/copy_input.tsx

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import * as React from 'react';
+import { EuiCopy, EuiButtonIcon, EuiFlexGroup, EuiFlexItem, EuiPanel, EuiText } from '@elastic/eui';
+
+export interface EndpointUrlProps {
+  value: string;
+}
+
+export const CopyInput: React.FC<EndpointUrlProps> = ({ value }) => {
+  const textRef = React.useRef<HTMLSpanElement>(null);
+
+  return (
+    <EuiPanel borderRadius="none" hasShadow={false} color={'subdued'} grow={false}>
+      <EuiFlexGroup gutterSize="s">
+        <EuiFlexItem>
+          <EuiText
+            size={'s'}
+            color={'subdued'}
+            style={{ wordBreak: 'break-all' }}
+            data-test-subj="copyInputValue"
+          >
+            <span
+              ref={textRef}
+              onMouseDown={(event) => {
+                const span = textRef.current;
+                if (!span) return;
+                if (window.getSelection && document.createRange) {
+                  const selection = window.getSelection();
+                  if (!selection) return;
+                  event.preventDefault();
+                  const range = document.createRange();
+                  range.selectNodeContents(span);
+                  selection.removeAllRanges();
+                  selection.addRange(range);
+                }
+              }}
+            >
+              {value}
+            </span>
+          </EuiText>
+        </EuiFlexItem>
+        <EuiFlexItem grow={false}>
+          <EuiCopy textToCopy={value}>
+            {(copy) => (
+              <EuiButtonIcon onClick={copy} iconType="copyClipboard" size="m" color={'text'} />
```

Comment:
Warning in console: 

button_icon.js:53 EuiButtonIcon requires aria-label or aria-labelledby to be specified because icon-only
      buttons are screen-reader-inaccessible without them.

Reply:
Good catch, done!.